pkg/monitoring/metrics/virt-controller/vmsnapshot.go

```diff
@@ -0,0 +1,41 @@
+package virt_controller
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	snapshotv1 "kubevirt.io/api/snapshot/v1beta1"
+)
+
+var (
+	vmSnapshotMetrics = []operatormetrics.Metric{
+		VmSnapshotCreationTimestamp,
+	}
+
+	VmSnapshotCreationTimestamp = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vmsnapshot_timestamp_seconds",
+			Help: "Unix timestamp of when the VM snapshot was created. Including the VM name, namespace, and snapshot name.",
+		},
+		[]string{"vm_name", "vm_namespace", "snapshot_name", "snapshot_namespace", "snapshot_uid"},
+	)
+)
+
+func NewVMSnapshotCreated(snapshot *snapshotv1.VirtualMachineSnapshot, content *snapshotv1.VirtualMachineSnapshotContent) {
+	if snapshot.Status != nil && snapshot.Status.CreationTime != nil {
+		VmSnapshotCreationTimestamp.WithLabelValues(
+			content.Spec.Source.VirtualMachine.Name,
+			content.Spec.Source.VirtualMachine.Namespace,
+			snapshot.Name,
+			snapshot.Namespace,
+			string(snapshot.UID),
+		).Set(float64(snapshot.CreationTimestamp.Unix()))
+	}
+}
+
+func GetVMSnapshotMetric(vm, vmNamespace, snapshot, snapshotNamespace, snapshotUid string) (*io_prometheus_client.Metric, error) {
```

Comment:
let's return the value instead of the raw Metric, like in https://github.com/kubevirt/kubevirt/blob/a3a9e83da6fff2de37b7fd87efb43466cd6e07d4/pkg/monitoring/metrics/virt-controller/vmi_metrics.go#L50

in this case we can return `(time.Time, error)`

Reply:
done hope it make sense the fix

Reply:
I think @machadovilaca meant for you to return return *dto.Gauge.Value, nil